### PR TITLE
Balder/allow multiple requests [run-systemtest]

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst or @havardpe PR

### CODE REVIEW

**Review by geirst — APPROVED** [review]
👍

File: config/src/vespa/config/frt/frtsource.cpp [hunk 1]
```diff
@@ -79,33 +81,48 @@ FRTSource::RequestDone(FRT_RPCRequest * request)
...
             return;
         LOG(spam, "Killing task");
         _task->Kill();
+        inflight = _inflight;
     }
     LOG(spam, "Aborting");
-    if (_currentRequest.get() != NULL)
-        _currentRequest->abort();
-    LOG(spam, "Syncing");
-    _connectionFactory->syncTransport();
```
Comment (havardpe) [inline config/src/vespa/config/frt/frtsource.cpp:107]: you probably want to keep this one to ensure the transport thread has exited this object after close is called (after waiting for inflight requests)

File: config/src/vespa/config/frt/frtsource.cpp [hunk 2]
```diff
@@ -79,33 +81,48 @@ FRTSource::RequestDone(FRT_RPCRequest * request)
...
-    _currentRequest.reset(0);
+    for (auto & request : inflight) {
+        std::move(request.second)->abort();
```
Comment (havardpe) [inline config/src/vespa/config/frt/frtsource.cpp:120]: as far as I can see, this move does nothing

File: config/src/vespa/config/frt/frtsource.cpp [hunk 3]
```diff
@@ -79,33 +81,48 @@ FRTSource::RequestDone(FRT_RPCRequest * request)
...
+    }
+    inflight.clear();
+    LOG(spam, "Waiting");
+    std::unique_lock guard(_lock);
+    while (!_inflight.empty()) _cond.wait(guard);
```
Comment (havardpe) [inline config/src/vespa/config/frt/frtsource.cpp:125]: please add some braces here

**Pull request merged**